include/fiction/technology/sidb_on_the_fly_gate_library.hpp

```diff
@@ -146,7 +166,7 @@ class sidb_on_the_fly_gate_library : public fcn_gate_library<sidb_technology, 60
      * @return Bestagon gate representation of `t` including mirroring.
      */
     template <typename GateLyt, typename CellLyt, typename Params>
-    static fcn_gate set_up_gate(const GateLyt& lyt, const tile<GateLyt>& t, const Params& parameters = Params{})
+    static fcn_gate set_up_gate(const GateLyt& lyt, const tile<GateLyt>& t, const Params& params)
```

Comment:
This is a really long function. Can it be broken down meaningfully?

Reply:
hmm, I know, and I am not a big fan of long functions either. However, I don't see a reasonable way to split it up. We could put everything inside `` if (lyt.is_buf(n))`` into another function, but then the question arises, why exactly for this one?